### PR TITLE
xfitter: runtime fixes

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

Fixes xfitter. It is a UB in C++ for function to declare a return value and not provide any, a patch was added to fix that. Also qcdnum 17-01-14 deprecates some of the functions used by xfitter, so downgrade it for now.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/physics/xfitter/undefined_behavior.patch [hunk 1]
```diff
@@ -1,5 +1,4 @@
 diff --git a/DY/src/finterface.cc b/DY/src/finterface.cc
-index 0405786..eb171d0 100644
 --- a/DY/src/finterface.cc
```
Comment (Mic92) [inline pkgs/applications/science/physics/xfitter/undefined_behavior.patch]: Have you also send this patch upstream?

**Comment by veprbl** [comment]
@GrahamcOfBorg build apfelgrid qcdnum

**Comment by veprbl** [comment]
@GrahamcOfBorg build apfelgrid qcdnum

File: pkgs/applications/science/physics/xfitter/undefined_behavior.patch (hunk 1 above)
Comment (Mic92) [reply pkgs/applications/science/physics/xfitter/undefined_behavior.patch]: Did you had a chance to submit the patch?

Comment (veprbl) [reply pkgs/applications/science/physics/xfitter/undefined_behavior.patch]: @Mic92  I did. Fingers crossed, will see how it goes.

**Pull request merged**